source/includes/Encounter/encounter.md

```diff
@@ -47,202 +41,747 @@ Consultation and Dispensing**.
     
     
-### List encounter by UUID.
+## List encounter by UUID.
 
-```console
+> List encounter by UUID
+
+```shell
 GET /encounter/:target_encounter_uuid
 ```
-    Retrieve an encounter by its UUID. Returns a `404 Not Found` status if Encounter not exists. If user not logged 
-    in to perform this action, a `401 Unauthorized` status returned.
+
+```java
+
+OkHttpClient client = new OkHttpClient().newBuilder()
+  .build();
+Request request = new Request.Builder()
+  .url("/openmrs/ws/rest/v1/encounter/d3360c01-9813-4ff8-bd81-909af6612632")
+  .method("GET", null)
+  .addHeader("Authorization", "Basic YWRtaW46QWRtaW4xMjM=")
+  .addHeader("Cookie", "JSESSIONID=3EFCD2FD54D00BE8491DFFD43AA706DC")
+  .build();
+Response response = client.newCall(request).execute();
+
+```
+
+```javascript
+
+var requestHeaders = new Headers();
+requestHeaders.append("Authorization", "Basic YWRtaW46QWRtaW4xMjM=");
+requestHeaders.append("Cookie", "JSESSIONID=3EFCD2FD54D00BE8491DFFD43AA706DC");
+
+var requestOptions = {
+  method: 'GET',
+  headers: requestHeaders,
+  redirect: 'follow'
+};
+
+fetch("/openmrs/ws/rest/v1/encounter/d3360c01-9813-4ff8-bd81-909af6612632", requestOptions)
+  .then(response => response.text())
+  .then(result => console.log(result))
+  .catch(error => console.log('error', error));
+
+```
+
+* Retrieve an encounter by its UUID. Returns a `404 Not Found` status if Encounter not exists. If user not logged in to perform this action, a `401 Unauthorized` status returned.
     
    
 ## Create an encounter
 
-```console
+> Creating a new Encouter in existing visit
+
+```shell
 POST /Encounter
 {
-  "encounterDatetime": "2019-10-16 12:08:43",
-  "patient": "070f0120-0283-4858-885d-a20d967729cf",
-  "encounterType": "e22e39fd-7db2-45e7-80f1-60fa0d5a4378",
-  "location": "aff27d58-a15c-49a6-9beb-d30dcfc0c66e",
+  "encounterDatetime": "2015-02-24T06:08:25.000+0000",
+  "patient": "96be32d2-9367-4d1d-a285-79a5e5db12b8",
+  "encounterType": "67a71486-1a54-468f-ac3e-7091a9a79584",
+  "location": "58c57d25-8d39-41ab-8422-108a0c277d98",
   "encounterProviders": [
     {
       "provider": "bb1a7781-7896-40be-aaca-7d1b41d843a6",
       "encounterRole": "240b26f9-dd88-4172-823d-4a8bfeb7841f"
     }
-  ]
+  ],
+  "visit":"69b3265b-1521-4c27-8d3f-ae99b689e7a9"
+}
+```
+
+> Creating a new Encouter for a new Visit
+
+```shell
+POST /Encounter
+{
+    "encounterDatetime": "2015-02-24T06:08:25.000+0000",
+    "patient": "96be32d2-9367-4d1d-a285-79a5e5db12b8",
+    "encounterType": "67a71486-1a54-468f-ac3e-7091a9a79584",
+    "location": "58c57d25-8d39-41ab-8422-108a0c277d98",
+    "encounterProviders": [
+        {
+            "provider": "bb1a7781-7896-40be-aaca-7d1b41d843a6",
+            "encounterRole": "240b26f9-dd88-4172-823d-4a8bfeb7841f"
+        }
+    ],
+    "visit": {
+        "patient": "96be32d2-9367-4d1d-a285-79a5e5db12b8",
+        "visitType": "7b0f5697-27e3-40c4-8bae-f4049abfb4ed",
+        "startDatetime": "2015-02-24T06:08:25.000+0000"
```

Comment:
@LuGO0 you need a stopDatetime here, otherwise the visit will not be closed, and you can have infinite visits

Reply:
okay @solemabrothers I will make suitable changes !!